src/profile-logic/profile-data.js

```diff
@@ -425,12 +399,29 @@ export function getTimingsForCallNodeIndex(
   sampleIndexOffset: number,
   categories: CategoryList
 ): TimingsForPath {
-  const { samples, stackTable, funcTable } = thread;
+  /* ------------ Variables definitions ------------*/
+
+  // This is the data from the filtered thread that we'll loop over.
```

Comment:
I also added comments on top of the variables we use here. Hopefully this should make the function more easily understandable.